ClothesApp/Application.UnitTests/AddressService/AddressServiceTests.cs

```diff
@@ -0,0 +1,150 @@
+using Application;
+using Application.Dtos.Addresses;
+using Application.Exceptions;
+using AutoMapper;
+using Bogus;
+using Domain.Entities;
+using Domain.Enums;
+using Infrastructure;
+using Infrastructure.Repositories;
+using UnitTests.AddressService.AddAddress;
+using UnitTests.AddressService.GetAddress;
+using UnitTests.AddressService.UpdateAddress;
+
+namespace UnitTests.AddressService;
+
+public class AddressServiceTests : BaseTest
+{
+    private Application.Services.AddressService CreateAddressService(ClothesAppContext context)
+    {
+        return new Application.Services.AddressService(
+            new AddressRepository(context),
+            new UserRepository(context),
+            new Mapper(new MapperConfiguration(cfg 
+                => cfg.AddProfile<AppMappingProfile>())));
+    }
+    
+    [Theory]
+    [ClassData(typeof(GetAddressesTestData))]
+    public async Task GetAddresses_WhereCalledWithCorrectData(GetAddressTestCase testCase)
+    {
+        var addressService = CreateAddressService(Context);
+        Context.Users.Add(testCase.User);
+        Context.Addresses.AddRange(testCase.Addresses);
+        await Context.SaveChangesAsync();
+        
+        var result = await addressService.GetAddresses(testCase.User.Id);
+        
+        result.Should().BeEquivalentTo(testCase.ExpectedResult);
+    }
+    
+    [Theory]
+    [ClassData(typeof(GetAddressesTestData))]
+    public async Task GetAddresses_WhereCalledWithUndiscoveredUserId_ThrowNotFound(GetAddressTestCase testCase)
+    {
+        var addressService = CreateAddressService(Context);
+        Context.Users.Add(testCase.User);
+        await Context.SaveChangesAsync();
+        
+        Func<Task> act = async () => await addressService.GetAddresses(testCase.User.Id - 1);
+
+        await act.Should().ThrowAsync<NotFoundException>();
+    }
+
+    private User GenerateUser()
```

Comment:
All such things can be in a base class or in other static class for all generated entities